agent/lib_mongodb.c

```diff
@@ -173,7 +175,166 @@ NR_PHP_WRAPPER(nr_mongodb_operation) {
 }
 NR_PHP_WRAPPER_END
 
-void nr_mongodb_enable(TSRMLS_D) {
+#else
+
+NR_PHP_WRAPPER(nr_mongodb_operation_before) {
+  (void)wraprec;
+  nr_segment_t* segment;
+  segment = nr_segment_start(NRPRG(txn), NULL, NULL);
+  segment->wraprec = auto_segment->wraprec;
```

Comment:
`nr_segment_start` can [return a null](https://github.com/newrelic/newrelic-php-agent/blob/main/axiom/nr_segment.c#L166) so we should check for null before trying to access the wraprec.

Reply:
added here 59e8ec2a240efc89b44251c7681e6081c562850c but after looking into this I can't find any instance where we bother checking this value for `NULL` in any other framework instrumentation. Drupal, Predis, even the non-OAPI mongodb implementation all assume this value won't be `NULL`.

Reply:
Seems like a gap introduced when we needed to set wraprec. Not checking could cause issues if a txn is ended midway.

Looking at some examples...

in old mongodb instrumentation, it didn't need a check because it didn't access segment and only passed it to segment_end which does the null check: https://github.com/newrelic/newrelic-php-agent/blob/main/agent/lib_mongodb.c#L163

drupal:
trying to access wraprec, should have a check before trying: https://github.com/newrelic/newrelic-php-agent/blob/main/agent/fw_drupal.c#L261

here, it doesn't need a check because we aren't trying to access segment elements like wraprec, and  the segment is only used when we call [segment_end](https://github.com/newrelic/newrelic-php-agent/blob/main/axiom/nr_segment.c#L708) which does check for null.

same thing in https://github.com/newrelic/newrelic-php-agent/blob/main/agent/php_internal_instrument.c#L527
If we don't access it and simply pass the segment to nr_segment_end or nr_php_pdo_end_segment_sql or nr_segment_datastore_end, all those check for null, so we don't need to check.

but if we want to access it, a null check is needed like here: https://github.com/newrelic/newrelic-php-agent/blob/main/agent/php_internal_instrument.c#L1311